src/frontend/src/optimizer/plan_node/logical_values.rs

```diff
@@ -132,20 +154,35 @@ impl ToBatch for LogicalValues {
 
 impl ToStream for LogicalValues {
     fn to_stream(&self, _ctx: &mut ToStreamContext) -> Result<PlanRef> {
-        Err(RwError::from(ErrorCode::NotImplemented(
-            "Stream values executor is unimplemented!".to_string(),
-            None.into(),
-        )))
+        Ok(StreamValues::new(self.clone()).into())
     }
 
     fn logical_rewrite_for_stream(
         &self,
         _ctx: &mut RewriteStreamContext,
     ) -> Result<(PlanRef, ColIndexMapping)> {
-        Err(RwError::from(ErrorCode::NotImplemented(
-            "Stream values executor is unimplemented!".to_string(),
-            None.into(),
-        )))
+        let row_id_index = self.schema().len();
+        let col_index_mapping = ColIndexMapping::identity_or_none(row_id_index, row_id_index + 1);
+        let ctx = self.ctx();
+        let mut schema = self.schema().clone();
+        schema.fields.push(Field {
+            data_type: DataType::Int64,
+            name: "_row_id".to_string(),
+            sub_fields: vec![],
+            type_name: "int64".to_string(),
+        });
+        let rows = self.rows().to_vec();
+        let row_with_id = (0..rows.len())
+            .zip_eq_fast(rows.into_iter())
+            .map(|(i, mut r)| {
+                r.extend_one(
+                    Literal::new(Some(ScalarImpl::Int64(i as i64)), DataType::Int64).into(),
+                );
+                r
+            })
+            .collect_vec();
```

Comment:
Simply use `.enumerate().update(..)`? BTW, `extend_one` is simply `push` here. 😄

Reply:
I forgot `enumerate()` 🤣